### DISCUSSION TITLE
Derive the SU(2) from the corresponding O(3) matrices under the appropriate basis, instead of using a list in the source code.

### DISCUSSION THREAD

**Discussion by hongyi-zhao** [goodluck1982/SpaceGroupIrep#28] [General]
I noticed that the following [list](https://github.com/goodluck1982/SpaceGroupIrep/blob/c9b1fcad6ff2a3e5ca7ec27b5dc5ad6225d6edde/SpaceGroupIrep.wl#L1986) is used for building the correspondence between the O(3) and SU(2):

```mathematica
(* Note that the matrix of C32- in BC book tab 6.1 and 6.7 are wrong. C34+ in the tab 6.1 of my BC
   book is also wrong, but the one in the pdf version is right. *)
SpinRot0=<|"E"->{{1,0},{0,1}}, "C2x"->{{0,-I},{-I,0}},
  "C2y"->{{0,-1},{1,0}}, "C2z"->{{-I,0},{0,I}},
  "C31+"->{{1+I,1-I},{-1-I,1-I}}/2, "C32+"->{{1+I,-1+I},{1+I,1-I}}/2,
  "C33+"->{{1-I,-1-I},{1-I,1+I}}/2, "C34+"->{{1-I,1+I},{-1+I,1+I}}/2,
  "C31-"->{{1-I,-1+I},{1+I,1+I}}/2, "C32-"->{{1-I,1-I},{-1-I,1+I}}/2,
  "C33-"->{{1+I,1+I},{-1+I,1-I}}/2, "C34-"->{{1+I,-1-I},{1-I,1-I}}/2,
  "C4x+"->{{1,-I},{-I,1}}/Sqrt[2], "C4y+"->{{1,1},{-1,1}}/Sqrt[2],
  "C4z+"->{{1+I,0},{0,1-I}}/Sqrt[2], "C4x-"->{{1,I},{I,1}}/Sqrt[2],
  "C4y-"->{{1,-1},{1,1}}/Sqrt[2], "C4z-"->{{1-I,0},{0,1+I}}/Sqrt[2],
  "C2a"->{{0,-1+I},{1+I,0}}/Sqrt[2], "C2b"->{{0,-1-I},{1-I,0}}/Sqrt[2],
  "C2c"->{{-I,I},{I,I}}/Sqrt[2], "C2d"->{{-I,-1},{1,I}}/Sqrt[2],
  "C2e"->{{-I,-I},{-I,I}}/Sqrt[2], "C2f"->{{I,-1},{1,-I}}/Sqrt[2],
  "C6+"->{{I+Sqrt[3],0},{0,-I+Sqrt[3]}}/2, "C3+"->{{1+I Sqrt[3],0},{0,1-I Sqrt[3]}}/2,
  "C2"->{{-I,0},{0,I}}, "C3-"->{{1-I Sqrt[3],0},{0,1+I Sqrt[3]}}/2,
  "C6-"->{{-I+Sqrt[3],0},{0,I+Sqrt[3]}}/2, "C21p"->{{0,-I},{-I,0}},
  "C22p"->{{0,I+Sqrt[3]},{I-Sqrt[3],0}}/2, "C23p"->{{0,I-Sqrt[3]},{I+Sqrt[3],0}}/2,
  "C21pp"->{{0,-1},{1,0}}, "C22pp"->{{0,-1+I Sqrt[3]},{1+I Sqrt[3],0}}/2,
  "C23pp"->{{0,-1-I Sqrt[3]},{1-I Sqrt[3],0}}/2|>;
```
Why not direct calculate/derive the SU(2) based on the group isomorphism between SU(2) and O(3)? BTW, I noticed a Mathematica package [here](https://renatofonseca.net/groupmath) and hope it can give some clues/hints/tips.

Regards,
HZ

**Comment by goodluck1982** [comment]
Now that we utilize the irep data in the BC book, i.e. Tab. 5.7 and 6.13, we have to conform to the convention in the BC book. The BC book defines such spin rotation matrices and we just follow them. As for why the authors of the BC book chose such definitions, we don't know either.

However, there indeed is freedom to name the spin rotation due to the two-to-one homomorphism from SU(2) to SO(3). It's known that, if g \in SU(2) maps to R \in SO(3), then -g also maps to R. For example, the BC book defines `{{0,-1},{1,0}}` as `C2y`, but in principle  `-{{0,-1},{1,0}}` can also be defined as `C2y`.

**Comment by hongyi-zhao** [comment]
I mean, on the basis of following the basic principles of the original book, improvement in the efficiency, logic, readability, and maintainability of the code. Believing everything in books is worse than having no books at all, and, as you have found, there are mistakes and flaws in the original book too.

As an example, I noticed that `gtpack` can display [the representations corresponding to O(3) and SU(2) conveniently](https://gtpack.org/tutorials/standard-representations-o3-su2-su2xs/): 

![image](https://user-images.githubusercontent.com/11155854/152792580-321d9b3e-5c62-49ce-99ba-374184beddd9.png)

BTW, I also noticed that the rotation matrix given by `gtpack` is not in the same format as the one used by BC book:

![image](https://user-images.githubusercontent.com/11155854/152794438-711abb12-efa4-4a01-987d-2ac95365cef8.png)

**Comment by goodluck1982** [comment]
I understand your meaning.
One of the purposes of `SpaceGroupIrep` package is to facilitate learning the BC book. The results output by the package are designed to be the same with the BC book (except the typos & mistakes in the BC book). To achieve this goal, we have to adopt the definitions in the BC book. 

For example, the following is the double-valued irep of SG 167 (in Tab.6.13 on page 537 of the BC book)
![图片](https://user-images.githubusercontent.com/48718961/152792626-e2fe853e-2761-4383-b126-5716b4c43b29.png)
The generators of "Z" point include {barC3+|001}. The spin rotation matrix of barC3+ is minus the one of C3+. Only we use the same definitions as in the BC book can we obtain the correct Herring's little group which isomorphic to the abstract group G_48^9 given in the Tab.5.1 of the BC book.

**Comment by goodluck1982** [comment]
### Why should they be in the same format?

### In fact, the difference lies in the bases they used. Yes, bases appear again!
The O(3) matrix that gtpack gives uses the Cartesian bases, while the BC book use the basic vectors of the primitive cell as bases whose merit is that all the matrix elements are integers.  And actually, if one only want to get the rotation matrix under Cartesian bases, he/she does not need any third-party package. The build-in `RotationMatrix` function of mathematica is enough.
![图片](https://user-images.githubusercontent.com/48718961/152796294-8c6d5fd9-c4a9-460c-90d7-19c32560ed0e.png)

Here are two pages from the ppt of my group-theory course for reference:
![图片](https://user-images.githubusercontent.com/48718961/152798296-266523ab-77e6-4588-a2a3-8519380f58ab.png)
![图片](https://user-images.githubusercontent.com/48718961/152798441-f9065697-a656-4f14-aeca-dfc004be660c.png)



### As for the spin rotation matrix, one can simply use the following expression [喀兴林《高等量子力学》]
![图片](https://user-images.githubusercontent.com/48718961/152796881-76beed22-5529-4d34-9ed7-df715678ab3c.png)
Also this can be obtained by the build-in functions of mathematica:
![图片](https://user-images.githubusercontent.com/48718961/152797454-840cff7b-9b99-4d02-9fb4-04acbdaa93ab.png)
One can see that the above matrix is different from the `"C3+"` in the BC book.
![图片](https://user-images.githubusercontent.com/48718961/152797834-b4681abb-4fd1-4931-962e-60f2eaf376fa.png)
But what's interesting is the `"C3+"` matrix in the BC book happens to be the same with the one from `gtpack`.

**Comment by hongyi-zhao** [comment]
> ### Why should they be in the same format?
> ### In fact, the difference lies in the bases they used. Yes, bases appear again!
> 
> The O(3) matrix that gtpack gives uses the Cartesian bases, while the BC book use the basic vectors of the primitive cell as bases whose merit is that all the matrix elements are integers.

Later, I thought about it for a second time and also understood the difference between them. One is based on the representation of basis in Cartesian coordinate system, and the other is corresponding to "Jones' faithful representation symbols" using fractional coordinate system, as you've described [here](https://github.com/goodluck1982/SpaceGroupIrep/issues/1#issuecomment-1019998632).

> ### As for the spin rotation matrix, one can simply use the following expression [喀兴林《高等量子力学》]
> 
> ![图片](https://user-images.githubusercontent.com/48718961/152796881-76beed22-5529-4d34-9ed7-df715678ab3c.png) Also this can be obtained by the build-in functions of mathematica: ![图片](https://user-images.githubusercontent.com/48718961/152797454-840cff7b-9b99-4d02-9fb4-04acbdaa93ab.png)

Sorry for my weak theoretical foundation. I wonder why don't you use the following computation method based on the expression shown in [喀兴林《高等量子力学》]:

```mathematica
PauliMatrix /@ {1, 2, 3}
MatrixExp[-I*(1/\[HBar])*\[Phi] ({n1, n2, n3} . %)] // 
  ExpToTrig // MatrixForm
% /. {\[Phi] -> 2 Pi/3, n1 -> 0, n2 -> 0, n3 -> 1} // MatrixForm
```
Though the above calculation will give a strange result as follows, but it seems more consistent to the formula given by the book [喀兴林《高等量子力学》]:

![image](https://user-images.githubusercontent.com/11155854/152912442-86d9a7ef-06c9-4c7c-89a6-f688d0241fab.png)

> But what's interesting is the `"C3+"` matrix in the BC book happens to be the same with the one from `gtpack`.

This is indeed a subtle underlying presentation problem that should be used self-consistently in a package.

**Comment by goodluck1982** [comment]
> > 
> 
> Sorry for my weak theoretical foundation. I wonder why don't you use the following computation method based on the expression shown in [喀兴林《高等量子力学》]:
> 
> ```mathematica
> PauliMatrix /@ {1, 2, 3}
> MatrixExp[-I*(1/\[HBar])*\[Phi] ({n1, n2, n3} . %)] // 
>   ExpToTrig // MatrixForm
> % /. {\[Phi] -> 2 Pi/3, n1 -> 0, n2 -> 0, n3 -> 1} // MatrixForm
> ```
> 
> Though the above calculation will give a strange result as follows, but it seems more consistent to the formula given by the book [喀兴林《高等量子力学》]:

The answer has been given above, see [here](https://github.com/goodluck1982/SpaceGroupIrep/issues/12#issuecomment-1031451164). 

BTW: there is extra hbar in your calculation, because 
![图片](https://user-images.githubusercontent.com/48718961/152918176-68e9ff48-7edc-4b47-95e8-d4147ffcdb5e.png)

**Comment by hongyi-zhao** [comment]
Thank you for pointing out my logical mistake. Another view is to look at the relationship above from [Lie algebra](https://math.stackexchange.com/a/2212907): Pauli matrices, after multiplication by `i`, are so-called infinitesimal generators. These are elements of the Lie algebra su(2) of SU(2) group, and for such elements X the exponential mapping `t↦exp(tX)` gives a one-parameter subgroup of the Lie group.

**Comment by goodluck1982** [comment]
You are right.
The following is captured from my ppt.
![image](https://user-images.githubusercontent.com/48718961/152947462-9b1c82d7-0268-4140-8d69-ea97702dfc14.png)

**Comment by hongyi-zhao** [comment]
I noticed that in `SpaceGroupIrep`, the rotation matrix is represented by a matrix corresponding to `O(3)`, which means that `det(O(3))=+/-1`, instead of the SO(3), which only includes the ones whose determinant is 1.

Side remark: It must be pointed out that Lie algebra is just one of the representation theories of SU(2), as explained [here](https://en.wikipedia.org/wiki/Representation_theory_of_SU(2)). In general, there are many group representation theories, just as there are many representation theories of quantum mechanics. They have their own advantages and disadvantages, so they are suitable for different fields. Researchers must choose according to their needs, and even invent new representation theories according to specific problems.

**Comment by goodluck1982** [comment]
All rotations in 3D space are just O(3), not only SO(3).

**Comment by hongyi-zhao** [comment]
Then, how to interpret the description [here](https://en.wikipedia.org/wiki/3D_rotation_group)?:

![image](https://user-images.githubusercontent.com/11155854/152985649-ae8a6566-e721-4f94-9906-d075f25d649b.png)

and [here](https://en.wikipedia.org/wiki/Orthogonal_group):

![image](https://user-images.githubusercontent.com/11155854/152985919-058e0c9f-b8e3-42ea-bfec-5cbab4651935.png)

**Comment by goodluck1982** [comment]
The "rotation" I mentioned above is a generalized concept which contains "proper rotation" (det = 1) and "improper rotation" (det=-1).  In spite of being called "rotation group", SO(3) only contains proper rotations. It is O(3) that contains both proper and improper rotations.  Mirror and inversion are both improper rotations.

**Comment by hongyi-zhao** [comment]
> Here are two pages from the ppt of my group-theory course for reference:
> ![图片](https://user-images.githubusercontent.com/48718961/152798296-266523ab-77e6-4588-a2a3-8519380f58ab.png)
> ![图片](https://user-images.githubusercontent.com/48718961/152798441-f9065697-a656-4f14-aeca-dfc004be660c.png)

Below are some other methods to prove the crystallographic restriction theorem:

https://mathworld.wolfram.com/CrystallographyRestriction.html
https://en.wikipedia.org/wiki/Crystallographic_restriction_theorem

**Comment by hongyi-zhao** [comment]
> ### As for the spin rotation matrix, one can simply use the following expression [喀兴林《高等量子力学》]
> 
> ![图片](https://user-images.githubusercontent.com/48718961/152796881-76beed22-5529-4d34-9ed7-df715678ab3c.png) Also this can be obtained by the build-in functions of mathematica: ![图片](https://user-images.githubusercontent.com/48718961/152797454-840cff7b-9b99-4d02-9fb4-04acbdaa93ab.png) One can see that the above matrix is different from the `"C3+"` in the BC book. ![图片](https://user-images.githubusercontent.com/48718961/152797834-b4681abb-4fd1-4931-962e-60f2eaf376fa.png) But what's interesting is the `"C3+"` matrix in the BC book happens to be the same with the one from `gtpack`.

I calculated all the SU(2) matrix representations, using the possible permutations of Pauli matrices as generators:

```mathematica
myfunc[pauli_] := 
 MatrixExp[-I*\[Phi] ({n1, n2, n3} . pauli)/2] /. {\[Phi] -> 2 Pi/3, 
     n1 -> 0, n2 -> 0, n3 -> 1} // ExpToTrig // MatrixForm
Table[{i, myfunc[PauliMatrix /@ i ]}, {i, 
   Permutations[{1, 2, 3}]}] // MatrixForm
```
![image](https://user-images.githubusercontent.com/11155854/153123657-027e48d0-aac0-45ef-b2ad-0b2044c7aa5d.png)

As you can see, only three sets of non-repetitive SU(2) matrix representations are obtained and none of them is equivalent to the one used by `SpaceGroupIrep`:

![image](https://user-images.githubusercontent.com/11155854/153143256-92dce456-6059-428b-abdd-ec3e3603b3bb.png)

I really want to find a method to construct all the SU(2) matrix representations corresponding to the O(3) matrices used in BC book. I think that the correspondence between O(3) and SU(2) representation must have a self-contained algorithm in the same monograph/tool/package.

**Comment by goodluck1982** [comment]
The best way to construct the SU(2) used the BC book is to refer to the BC book, although it cannot solve all problems.
![image](https://user-images.githubusercontent.com/48718961/153129252-50e17834-6097-42ae-8b61-f6256813264d.png)
![image](https://user-images.githubusercontent.com/48718961/153133806-866c2c0e-11a0-47c4-a757-851b8aee3fe0.png)

In fact, the SU(2) matrix in the BC book uses the bases {spin-down, spin-up}, which is different from the usually used {spin-up, spin-down} as used by 
![img](https://user-images.githubusercontent.com/48718961/152918176-68e9ff48-7edc-4b47-95e8-d4147ffcdb5e.png)
This point has been mentioned in my paper 
![image](https://user-images.githubusercontent.com/48718961/153134782-689f4438-b3b9-45d8-9918-6b8b1c1a0c12.png)

A similarity transformation by the matrix {{0,1},{1,0}} can relate the two equations. This actually is a **bases** issue again!

**Comment by hongyi-zhao** [comment]
> The best way to construct the SU(2) used the BC book is to refer to the BC book, although it cannot solve all problems.

Do you mean the cases of improper rotations by saying "although it cannot solve all problems", as you have commented [here](https://github.com/goodluck1982/SpaceGroupIrep/issues/12#issuecomment-1032562905)?

**Comment by goodluck1982** [comment]
> Do you mean the cases of improper rotations by saying "although it cannot solve all problems", as you have commented [here](https://github.com/goodluck1982/SpaceGroupIrep/issues/12#issuecomment-1032562905)?

No. I mean the BC book only defines its own convention, but it dose not say much about its relation to other conventions, e.g. about the SU(2) matrix.

**Comment by hongyi-zhao** [comment]
Anyway, though the BC book is the world-famous monograph, I think this is not a good style/habit, especially for researchers working in natural sciences.

**Comment by goodluck1982** [comment]
Although it's not good, it's inevitable and ubiquitous in scientific community. Researchers usually favor to use concepts/terminology/convention for their taste.

**Comment by hongyi-zhao** [comment]
> In fact, the SU(2) matrix in the BC book uses the bases {spin-down, spin-up}, which is different from the usually used {spin-up, spin-down} as used by ![img](https://user-images.githubusercontent.com/48718961/152918176-68e9ff48-7edc-4b47-95e8-d4147ffcdb5e.png) This point has been mentioned in my paper ![image](https://user-images.githubusercontent.com/48718961/153134782-689f4438-b3b9-45d8-9918-6b8b1c1a0c12.png)
> 
> A similarity transformation by the matrix {{0,1},{1,0}} can relate the two equations. This actually is a **bases** issue again!

I rechecked the related description in [the book written by the developers of `gtpack`](https://hk1lib.org/book/3514813/2653cf), and found the following explanation on page 29:

![image](https://user-images.githubusercontent.com/11155854/153817605-7e114e5e-529f-4975-9673-c79000bab084.png)

As you can see, `gtpack` uses the matrix `−u`, i.e, `-R` mentioned above,  as the standard. This is consistent with the results calculated below:

![image](https://user-images.githubusercontent.com/11155854/153818437-2eb5bf92-2111-4e9a-9172-e17f411e610d.png)

**Comment by goodluck1982** [comment]
Is strange. This is different from the result above https://github.com/goodluck1982/SpaceGroupIrep/issues/12#issuecomment-1031393364
![图片](https://user-images.githubusercontent.com/48718961/153823592-30f4732f-8828-4d8c-9ea2-74ce5a167324.png)

**Comment by hongyi-zhao** [comment]
I've also noticed this strange problem, as shown below:

![image](https://user-images.githubusercontent.com/11155854/153823961-5b796040-b259-47ff-a196-df6049a28f85.png)

And asked the author of `gtpack` about this and some related questions. The following is the full content of my email for your reference:

> 
> Some problems when using your gtpack package.
> 
> from: Hongyi Zhao <hongyi.zhao@gmail.com>
> to: wolfram.hergert@physik.uni-halle.de
> 
> Dear Prof. Dr. Wolfram Hergert,
> 
> I'm very glad to learn and use your wonderful gtpack package. However,
> I also encountered some problems in use, as follows:
> 
> 1. gtpack can't be reloaded repeatedly, even I've wiped my
> variables/functions and removed all the symbols from the kernel with
> the following commands, as commented here [1]:
> 
> ClearAll[Evaluate[$Context<>"*"]];
> Remove[Evaluate[$Context<>"*"]];
> 
> 
> Basically, I mean, if I try to run the following command sequences:
> 
>   << GroupTheory`;

>   ClearAll[Evaluate[$Context <> "*"]];

>   Remove[Evaluate[$Context <> "*"]];

>   << GroupTheory`
> 
> The following error will be triggered:
> 
> Error: The axis label x already exists!
> $Aborted
> 
> 
> 2. As we all know, when a user enters a partial string at the
> beginning of a command, Mathematica can prompt the user for all
> possible candidate commands. But I can't see this feature for all
> commands/functions defined in gtpack package. Considering that there
> are many custom commands and function names in the package, this is
> very inconvenient for users.
> 
> 3. I've noticed that the "SU(2)" matrix has the following two different forms.
> 
> When running the following command:
> 
> GTSU2Matrix[2 Pi/3, {0, 0, 1}] // MatrixForm
> 
> The result is as follows:
> 
> \left(
> \begin{array}{cc}
>  -\frac{1}{2}+\frac{i \sqrt{3}}{2} & 0 \\
>  0 & -\frac{1}{2}-\frac{i \sqrt{3}}{2} \\
> \end{array}
> \right)
> 
> 
> When running the following command:
> 
> GTChangeRepresentation["SU(2)"]
> GTGetMatrix[C3z] // MatrixForm
> 
> The result is as follows:
> 
> \left(
> \begin{array}{cc}
>  \frac{1}{2}+\frac{i \sqrt{3}}{2} & 0 \\
>  0 & \frac{1}{2}-\frac{i \sqrt{3}}{2} \\
> \end{array}
> \right)
> 
> Please refer to the attachment for details.
> 
> Any hints for these differences and questions will be highly appreciated.
> 
> [1] https://mathematica.stackexchange.com/a/884
> 
> Regards,
> HZ

**Reply by hongyi-zhao** [reply]
See the following reply given by `gtpack` developers:

```
Dear Prof Zhao,


Thanks for your question.


> 1. gtpack can't be reloaded repeatedly, even I've wiped my
> variables/functions and removed all the symbols from the kernel with
> the following commands, as commented here [1]:


I recommend using "Needs["GroupTheory`"]" instead of Get (or "<<"). To reload everything, I usually quit the Mathematica kernel. 


> 2. As we all know, when a user enters a partial string at the
> beginning of a command, Mathematica can prompt the user for all
> possible candidate commands. But I can't see this feature for all
> commands/functions defined in gtpack package. Considering that there
> are many custom commands and function names in the package, this is
> very inconvenient for users.


That's a good point! We will look into it. However, to obtain an overview of all implemented commands you can use


?GT*


> 3. I've noticed that the "SU(2)" matrix has the following two different forms.


It is important to mention that for SU(2) we obtain 4 matrices mediating three-fold rotations about the z-axis. We denote them by C3z, C3zi, DC3z, DC3zi (double group).

Please find attached a picture. Unfortunately, the GTSU2Matrix command seems to involve an overall factor of -1 which is not consistent with it's definition. I will repair this for the next version of GTPack. Thanks for spotting a bug! (e.g. a rotation about 2pi should give minus the identity matrix, which is not the case right now).


I hope these comments help. Please let me know if you have further questions.


Kind regards

Matthias Geilhufe
```
![image](https://user-images.githubusercontent.com/11155854/154930032-decbdea8-029e-46ff-8c00-41b6e9e44427.png)

**Comment by goodluck1982** [comment]
I see, the above "C3z" is the same with the "C3+" in the BC book, not directly from the equation ![img](https://user-images.githubusercontent.com/48718961/152918176-68e9ff48-7edc-4b47-95e8-d4147ffcdb5e.png)

I guess, maybe `gtpack` also uses the named SU(2) matrices in the BC book directly.

**Reply by hongyi-zhao** [reply]
As for SU(2) representation, there are two popular gauges: Pauli and Cartan. The `gtpack` uses the Pauli gauge, which is stated in its book.

**Reply by goodluck1982** [reply]
What are the detailed contents of the two gauges?

**Reply by hongyi-zhao** [reply]
The related description in [`gtpack`'s book](https://sg1lib.org/book/3514813/2653cf):
![image](https://user-images.githubusercontent.com/11155854/154968180-07030aa4-5189-4712-91b9-d78eb2796ea9.png)
![image](https://user-images.githubusercontent.com/11155854/154968463-3d349244-6a65-4e45-b0cd-de7b7ef2c2a7.png)
![image](https://user-images.githubusercontent.com/11155854/154968610-eda6f21d-bf44-4bfe-a223-1894fcbbc32b.png)

Related description in the [book](https://sg1lib.org/book/505244/b3697e) referred by `gtpack`:

![image](https://user-images.githubusercontent.com/11155854/154967562-ed39ce18-9e74-47b3-8a6c-8994b567072c.png)

**Reply by goodluck1982** [reply]
I have to say that it's my first time to know the Cartan gauge. What I have been using is just the Pauli gauge, because it's very natural that inversion has no impact on the axial vector spin, although I did not know its name.

**Reply by hongyi-zhao** [reply]
Other supplementary or more detailed explanations:

1. The following is from the explanation in [this book](https://hk1lib.org/book/503772/d7b672):

![image](https://user-images.githubusercontent.com/11155854/155047993-f944d656-80bf-4411-8fc5-d41e74d6c53e.png)
![image](https://user-images.githubusercontent.com/11155854/155048008-f66ca935-19f0-4cdc-84ac-ff2bbd87f3a6.png)
![image](https://user-images.githubusercontent.com/11155854/155048021-bc83f7ea-c586-48b2-8c8c-5f3416c0bc59.png)
![image](https://user-images.githubusercontent.com/11155854/155048029-aaef9981-0124-4a6f-b02f-5aeec2d31e2b.png)

2. Please also see some comments [here](https://physics.stackexchange.com/questions/329374/what-is-the-difference-between-a-pauli-spinor-a-weyl-spinor-and-a-cartan-spino) and [here](https://link.springer.com/content/pdf/bbm:978-3-540-29082-7/1.pdf).

**Reply by goodluck1982** [reply]
Thank you for your information!
